auto_submit/test/requests/check_pull_request_test.dart

```diff
@@ -297,47 +377,192 @@ void main() {
+      githubService.successMergeData = successMergeMock;
+      githubService.createCommentData = createCommentMock;
+      config = FakeConfig(githubService: githubService, githubGraphQLClient: githubGraphQLClient);
+      checkPullRequest = CheckPullRequest(config: config, pubsub: pubsub);
+
+      prRequested1 = PullRequestHelper(
+        authorAssociation: 'MEMBER',
+        reviews: <PullRequestReviewHelper>[
+          const PullRequestReviewHelper(
+              authorName: 'some_rando', state: ReviewState.APPROVED, memberType: MemberType.MEMBER)
+        ],
+        lastCommitStatuses: const <StatusHelper>[
+          StatusHelper.flutterBuildSuccess,
+        ],
+      );
+
+      final List<Response> responses = await checkPullRequest.get();
+      expectedOptions.add(flutterOption);
+      _verifyQueries(expectedOptions);
+      final String resBody = await responses[0].readAsString();
+      expect(resBody, 'Remove the autosubmit label for commit: ${pullRequest.head!.sha}.');
       assert(pubsub.messagesQueue.isEmpty);
     });
 
     test('Merges only _kMergeCountPerRepo PR per cycle per repo', () async {
-      final PullRequest pullRequest1 = generatePullRequest(prNumber: 22, repoName: 'flutter', login: 'flutter');
-      final PullRequest pullRequest2 = generatePullRequest(prNumber: 23, repoName: 'flutter', login: 'flutter');
-      final PullRequest pullRequest3 = generatePullRequest(prNumber: 24, repoName: 'cocoon', login: 'flutter');
-      config = FakeConfig(githubService: githubService);
+      final PullRequest pullRequest1 = generatePullRequest(prNumber: 0, repoName: 'flutter', login: 'flutter');
+      final PullRequest pullRequest2 = generatePullRequest(prNumber: 1, repoName: 'flutter', login: 'flutter');
+      final PullRequest pullRequest3 = generatePullRequest(prNumber: 2, repoName: cocoonRepo, login: 'flutter');
+      config = FakeConfig(githubService: githubService, githubGraphQLClient: githubGraphQLClient);
       checkPullRequest = CheckPullRequest(config: config, pubsub: pubsub);
       final Map<String, Set<PullRequest>> repoPullRequestsMap = <String, Set<PullRequest>>{
         'flutter/flutter': <PullRequest>{pullRequest1, pullRequest2},
         'flutter/cocoon': <PullRequest>{pullRequest3}
       };
 
       List<Map<int, String>> mergeResult = await checkPullRequest.checkPullRequests(repoPullRequestsMap);
-      expect(mergeResult[0], <int, String>{22: 'merged'});
-      expect(mergeResult[1], <int, String>{23: 'queued'});
-      expect(mergeResult[2], <int, String>{24: 'merged'});
+      expect(mergeResult[0], <int, String>{0: 'merged'});
+      expect(mergeResult[1], <int, String>{1: 'queued'});
+      expect(mergeResult[2], <int, String>{2: 'merged'});
       expect(pubsub.messagesQueue.length, 1);
       pubsub.messagesQueue.clear();
     });
   });
 }
+
+enum ReviewState {
+  APPROVED,
+  CHANGES_REQUESTED,
+}
+
+enum MemberType {
+  OWNER,
+  MEMBER,
+  OTHER,
```

Comment:
CONTRIBUTOR?

Reply:
Besides OWNER and MEMBER, author association can be other values including CONTRIBUTOR, COLLABORATOR, FIRST_TIMER, MANNEQUIN, etc. I used OTHER here to represent all other possible values.